wallet/src/test/java/com/android/identity_credential/wallet/OpenID4VPTest.kt

```diff
@@ -0,0 +1,136 @@
+                    "\"format\":{\"mso_mdoc\":{\"alg\":[\"ES256\",\"ES384\",\"ES512\",\"EdDSA\",\"ESB256\",\"ESB320\",\"ESB384\",\"ESB512\"]}}," +
+                    "\"constraints\":{\"fields\":[{\"path\":[\"\$['eu.europa.ec.eudiw.pid.1']['age_over_18']\"],\"intent_to_retain\":false}]}}]," +
+                "\"id\":\"32f54163-7166-48f1-93d8-ff217bdb0653\"}"),
+            authRequest.presentationDefinition
+        )
+        Assert.assertEquals("verifier-backend.eudiw.dev", authRequest.clientId)
+        Assert.assertEquals("9b4f04a3-8328-4a87-b0f1-2350f62d4736", authRequest.nonce)
+        Assert.assertEquals("https://verifier-backend.eudiw.dev/wallet/direct_post",
+            authRequest.responseUri)
+        Assert.assertEquals("44zZ7FmrM4nSuo5cJoQX2Gx1pKioAwVbN_gOaSOR1N-GGi2JfCky641R1UsSJT6bK2ALO4zUQ7SOVyc0mdy5kw",
+            authRequest.state)
+        Assert.assertEquals(parseToJsonElement(
+                "{\"authorization_encrypted_response_alg\":\"ECDH-ES\"," +
+                "\"authorization_encrypted_response_enc\":\"A128CBC-HS256\"," +
+                "\"id_token_encrypted_response_alg\":\"RSA-OAEP-256\"," +
+                "\"id_token_encrypted_response_enc\":\"A128CBC-HS256\"," +
+                "\"jwks_uri\":\"https://verifier-backend.eudiw.dev/wallet/jarm/44zZ7FmrM4nSuo5cJoQX2Gx1pKioAwVbN_gOaSOR1N-GGi2JfCky641R1UsSJT6bK2ALO4zUQ7SOVyc0mdy5kw/jwks.json\"," +
+                "\"subject_syntax_types_supported\":[\"urn:ietf:params:oauth:jwk-thumbprint\"]," +
+                "\"id_token_signed_response_alg\":\"RS256\"}"),
+            authRequest.clientMetadata!!)
+
+        val presentationSubmission = createPresentationSubmission(authRequest)
+        Assert.assertEquals("32f54163-7166-48f1-93d8-ff217bdb0653",
+            presentationSubmission.definitionId)
+        val descriptorMaps = presentationSubmission.descriptorMaps
+        for (descriptorMap: DescriptorMap in descriptorMaps) {
+            Assert.assertEquals("eu.europa.ec.eudiw.pid.1", descriptorMap.id)
+            Assert.assertEquals("mso_mdoc", descriptorMap.format)
+            Assert.assertEquals("$", descriptorMap.path)
+        }
+
+        val docRequest = formatAsDocumentRequest(authRequest.presentationDefinition["input_descriptors"]!!.jsonArray[0].jsonObject)
+        val expectedRequestedElems = listOf(
+            DocumentRequest.DataElement("eu.europa.ec.eudiw.pid.1", "age_over_18", false))
+        Assert.assertEquals(expectedRequestedElems, docRequest.requestedDataElements)
+    }
+
+    @Test
+    fun annexBJwtToPresentationSubmission() {
+        val authRequest = getAuthRequestFromJwt(SignedJWT.parse(annexBRequestObject), "example.com ")
+        Assert.assertEquals(parseToJsonElement(
+            "{\"input_descriptors\":" +
+                    "[{\"id\":\"org.iso.18013.5.1.mDL \"," +
+                    "\"format\":{\"mso_mdoc\":{\"alg\":[\"ES256\",\"ES384\",\"ES512\",\"EdDSA\",\"ESB256\",\"ESB320\",\"ESB384\",\"ESB512\"]}}," +
```

Comment:
I'm guessing ESB256 is ECDSA w/ SHA-256 on brainpoolP256r1? Ditto for the other ESB curves. Where is this defined, I don't see it in https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms

Reply:
yup, it does - the definitions are indicated in ISO 18013-7 Section B.6.2 Cryptographic curves

Reply:
I think they need to be defined in IANA too. I asked the ISO editor.